packages/xo-server/src/api/xosan.js

```diff
@@ -1138,8 +1139,12 @@ async function _prepareGlusterVm (
   await xapi.startVm(newVM)
   log.debug(`waiting for boot of ${ip}`)
   // wait until we find the assigned IP in the networks, we are just checking the boot is complete
+  // fix #3688
   const vmIsUp = vm =>
-    Boolean(vm.$guest_metrics && includes(vm.$guest_metrics.networks, ip))
+    Boolean(
+      vm.guest_metrics !== NULL_REF &&
+        includes(xapi.getObject(vm.guest_metrics).networks, ip)
+    )
   const vm = await xapi._waitObjectState(newVM.$id, vmIsUp)
```

Comment:
The issue is that you are monitoring an object (`newVM`) for changes in another objects (`newVM.$guest_metrics`), that's why it does not work.

You could do something like this:

```js
// wait for the VM to have guest metrics
newVM = await xapi._waitObjectState(newVM.$id, _ => _.guest_metrics !== NULL_REF)

// wait for the guest metrics to list the required IP address
await xapi._waitObjectState(newVM.guest_metrics, _ => includes(_.networks, ip))
```
```